querydsl-codegen/src/main/java/com/querydsl/codegen/EntityType.java

```diff
@@ -66,14 +75,13 @@ public EntityType(Type type) {
      *
      * @param type
      * @param superTypes
+     * @param variableNameFunction
      */
-    public EntityType(Type type, Set<Supertype> superTypes) {
+    public EntityType(Type type, Set<Supertype> superTypes, Function<EntityType, String> variableNameFunction) {
```

Comment:
I have tried to do the transformation in `EntitySerializer.introDefaultInstance`
```
String simpleName = !defaultName.isEmpty() ? defaultName : model.getModifiedSimpleName();
```
 but this way we were missing the logic in `EntityType` which mutates the `uncapSimpleName` before adding a `Property`
```
private Property validateField(Property field) {
        if (field.getName().equals(modifiedSimpleName) || field.getEscapedName().equals(modifiedSimpleName)) {
            do {
                modifiedSimpleName = StringUtils.uncapitalize(getType().getSimpleName()) + (escapeSuffix++);
            } while (propertyNames.contains(modifiedSimpleName));
        }
        return field;
    }
```
I found that the `EntityType` constructor was the most sensible place to make the transformation. The only problem being that `EntityType` is not a dependency injection managed class, so we have to change the code that instantiates it.